pkgs/stdenv/generic/default.nix

```diff
@@ -223,7 +223,7 @@ let
         # include it in the result, it *is* available to nix-env for queries.
         meta = { }
             # If the packager hasn't specified `outputsToInstall`, choose a default,
-            # namely `p.bin or p.out or p`;
+            # namely `lib.getBin p`;
```

Comment:
When `outs = [ "foo" ...]`, `pkg == pkg.foo` (except for `.outputUnspecified`), and consequently `hasOutput "foo"` will succeed and `findFirst` will never proceed further.

This whole discussion only strengthens my conviction that the code should be rewritten to use `getBin`...